iModelCore/ECDb/ECDb/RelationshipClassMap.cpp

```diff
@@ -1255,8 +1285,20 @@ void RelationshipClassLinkTableMap::AddIndex(SchemaImportContext& schemaImportCo
                 GenerateIndexColumnList(columns, sourceECInstanceIdColumn, sourceECClassIdColumn, targetECInstanceIdColumn, targetECClassIdColumn, nullptr);
                 break;
             case RelationshipIndexSpec::SourceAndTargetAndClassId:
+                {
                 GenerateIndexColumnList(columns, sourceECInstanceIdColumn, sourceECClassIdColumn, targetECInstanceIdColumn, targetECClassIdColumn, classId);
+                
+                // ECDb Profile version 4.0.0.5 onwards, update the unique index `uix_bis_ElementRefersToElements_sourcetargetclassid` to `uix_bis_ElementRefersToElements_sourcetargetclassid_memberpriority`
+                if (name.Equals("uix_bis_ElementRefersToElements_sourcetargetclassid"))
+                    {
+                    if (auto memberPriorityColumn = GetPrimaryTable().FindColumn("MemberPriority"); memberPriorityColumn)
+                        {
+                        columns.push_back(memberPriorityColumn);
+                        name.append("_memberpriority");
+                        }
+                    }

```

Comment:
Same here the index should be driven by the mapping CAs applied to the schema.  Doing this not only decouples the change from BisCore but it allows you to write a synthetic test in ECDb that doesn't use BisCore